AzureSpatialAnchorsSample/Assets/Scripts/SpatialAnchorsSample.cs

```diff
@@ -24,128 +24,202 @@ public class SpatialAnchorsSample : MonoBehaviour
         [Tooltip("The prefab used to represent an anchored object.")]
         private GameObject m_sampleSpatialAnchorPrefab = null;
 
-        [SerializeField]
-        [Tooltip("SpatialAnchorManager instance to use for this demo. This is required.")]
-        private SpatialAnchorManager m_spatialAnchorManager = null;
+        /// <summary>
+        /// Used to manage Azure Spatial Anchor sessions and queries in a Unity scene
+        /// </summary>
+        private SpatialAnchorManager m_cloudSpatialAnchorManager;
+
+        /// <summary>
+        /// The SpatialAnchorManager must have its session created after the Start() method.
+        /// Instead, we create it if it has not been created while starting the session.
+        /// </summary>
+        private bool m_spatialAnchorManagerSessionCreated = false;
+
+        /// <summary>
+        /// All anchor GameObjects found by or manually created in this demo. Used to ensure these are all
+        /// destroyed when the session is stopped.
+        /// </summary>
+        private List<GameObject> m_foundOrCreatedAnchorObjects = new List<GameObject>();
+
+        /// <summary>
+        /// If a new cloud session is started after we have previously saved cloud anchors, this spatial
+        /// anchor watcher will rediscover them.
+        /// </summary>
+        private CloudSpatialAnchorWatcher m_cloudSpatialAnchorWatcher;
 
-        private List<GameObject> m_foundOrCreatedAnchorObjects = new List<GameObject>(); // list of created or found anchors
+        /// <summary>
+        /// The Ids of all spatial anchors created during this demo. Used to when creating a watcher to
+        /// re-discover these anchors after stopping and starting the cloud session.
+        /// </summary>
         private List<string> m_createdSpatialAnchorIDs = new List<string>();
```

Comment:
are these "cloud anchor ids"?  if "cloud" then say it in the name.

Reply:
good point. Renamed to `m_cloudSpatialAnchorIDs`